### PR TITLE
Bpk 886 calendar focus tracking

### PR DESCRIPTION
**Pull request by georgegillams**
+ [x] For any new components I've made sure that the style can be extended by the consumer using a `className` override.
+ [x] For any new files I've included the [Apache 2.0 License header](https://github.com/Skyscanner/backpack/blob/master/packages/bpk-tokens/formatters/license-header.js#L2-L16) at the top of the file.
+ [x] I've updated `changelog.md` for any changes that need to be highlighted in the changelog.
+ [x] If I've updated `npm-shrinkwrap.json` those changes are intentional.

### CODE REVIEW

File: packages/bpk-component-calendar/src/Week.js [hunk 1]
```diff
@@ -87,6 +102,30 @@ class Week extends Component {
...
+  onMouseClick(date) {
+    this.props.onDateClick(date);
+    this.setState({ savedDate: null });
+  }
+
+  onMouseDown(date, month) {
+    if (isSameMonth(date, month)) {
+      this.props.onDateMouseDown(null, { date });
+    } else {
+      this.setState({ savedDate: date });
```
Comment (k0nserv) [inline packages/bpk-component-calendar/src/Week.js]: I'm trying to understand why this should be different. Can you expand on the reason?

File: packages/bpk-component-calendar/src/BpkCalendarGrid-test.js [hunk 2]
```diff
@@ -129,4 +129,29 @@ describe('BpkCalendarGrid', () => {
...
+      formatMonth={formatMonth}
+      formatDateFull={formatDateFull}
+      DateComponent={BpkCalendarDate}
+      daysOfWeek={weekDays}
+      weekStartsOn={0}
+      showWeekendSeparator
+      onDateMouseDown={onDateMouseDown}
+    />);
+
+    expect(onDateMouseDown.mock.calls.length).toBe(0);
```
Comment (k0nserv) [inline packages/bpk-component-calendar/src/BpkCalendarGrid-test.js]: This is redundant really

Comment (georgegillams) [reply packages/bpk-component-calendar/src/BpkCalendarGrid-test.js]: You mean checking that the `length` is `0`? or the whole test?

File: packages/bpk-component-calendar/src/Week.js [hunk 3]
```diff
@@ -87,6 +102,30 @@ class Week extends Component {
...
+    }
+  }
+
+  onMouseUp(event) {
+    if (this.state.savedDate != null) {
+      this.props.onDateMouseDown(event, { date: this.state.savedDate });
```
Comment (matthewdavidson) [inline packages/bpk-component-calendar/src/Week.js]: Calling a "mouse down" event handler on a mouse up event is surprising.

Comment (georgegillams) [reply packages/bpk-component-calendar/src/Week.js]: I agree - the 'mouseDown` and 'onClick' events should really be renamed if we are going with this solution, but that would presumably break a lot of stuff!

Comment (matthewdavidson) [reply packages/bpk-component-calendar/src/Week.js]: Yup, but I mean more fundamentally... they are not the same right? It's surprising that to bind a callback to a `mousedown` event for it to be called on `mouseup`.

Comment (k0nserv) [reply packages/bpk-component-calendar/src/Week.js]: That's a good point, maybe they sould have abstract names in week and above?

Comment (georgegillams) [reply packages/bpk-component-calendar/src/Week.js]: Something like `onDateFocused` and `onDateSelected`???

File: packages/bpk-component-calendar/src/BpkCalendarGrid-test.js (hunk 2 above)
Comment (k0nserv) [reply packages/bpk-component-calendar/src/BpkCalendarGrid-test.js]: Checking that the length is zero. That's guaranteed by the mock implementation

File: packages/bpk-component-calendar/src/Week.js (hunk 3 above)
Comment (k0nserv) [reply packages/bpk-component-calendar/src/Week.js]: Yeah something to that effect

**Pull request closed without merging**